DMX.Gatekeeper.Api.Tests.Unit/Services/Foundations/LabCommands/LabCommandServiceTests.Exception.RetrieveById.cs

```diff
@@ -0,0 +1,151 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ---------------------------------------------------------------
+
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using DMX.Gatekeeper.Api.Models.LabCommands;
+using DMX.Gatekeeper.Api.Models.LabCommands.Exceptions;
+using FluentAssertions;
+using Moq;
+using RESTFulSense.Exceptions;
+using Xeptions;
+using Xunit;
+
+namespace DMX.Gatekeeper.Api.Tests.Unit.Services.Foundations.LabCommands
+{
+    public partial class LabCommandServiceTests
+    {
+        [Theory]
+        [MemberData(nameof(CriticalDependencyException))]
+        public async Task ShouldThrowCriticalDependencyExceptionOnRetrieveIfCriticalErrorOccursAndLogItAsync(
+            Xeption criticalDependencyException)
+        {
+            // given
+            var someLabCommandId = Guid.NewGuid();
+
+            var failedLabCommandDependencyException =
+                new FailedLabCommandDependencyException(criticalDependencyException);
+
+            var expectedLabCommandDependencyException =
+                new LabCommandDependencyException(failedLabCommandDependencyException);
+
+            this.dmxApiBrokerMock.Setup(broker =>
+                broker.GetLabCommandByIdAsync(It.IsAny<Guid>()))
+                    .ThrowsAsync(criticalDependencyException);
+
+            // when 
+            ValueTask<LabCommand> retrieveLabCommandByIdTask =
+                this.labCommandService.RetrieveLabCommandByIdAsync(someLabCommandId);
+
+            LabCommandDependencyException actualLabCommandDependencyException =
+                await Assert.ThrowsAsync<LabCommandDependencyException>(
+                    retrieveLabCommandByIdTask.AsTask);
+
+            // then
+            actualLabCommandDependencyException.Should().BeEquivalentTo(
+                expectedLabCommandDependencyException);
+
+            this.dmxApiBrokerMock.Verify(broker =>
+                broker.GetLabCommandByIdAsync(someLabCommandId),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogCritical(It.Is(SameExceptionAs(
+                    expectedLabCommandDependencyException))),
+                        Times.Once);
+
+            this.dmxApiBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
```

Comment:
Where's handling the `HttpResponseNotFoundException`?
Where's handling `HttpResponseBadRequestException`?